Adding a proposal to bring some changes in the contributing guidelines. (#71)
johko/computer-vision-course: README.md

```
.
└── course/
    └── chapters/
        ├── chapter0/
        │   ├── introduction.mdx
        │   └── getting_started.mdx
        ├── chapter1/
        │   └── ...mdx
        └── _toctree.yml
```

1. First go to the [discussion section](https://github.com/johko/computer-vision-course/discussions/).
2. Each new chapter outline should be put up under a pined dicussion. As shown as in this [example](https://github.com/johko/computer-vision-course/discussions/80). 
3. Each chapter can contain multiple topics and sub-topics. All of that should be defined in the chapter outline. Also, for each sub-topics, a single contributor should be assigned to work on that. See more in this [example](https://github.com/johko/computer-vision-course/discussions/80).
4. Interested contributors motivated to work in a common chapter can come along togather to form a team. Where each contributor can directly contribute to a sub-topic and other fellow teammates can act as reviewers.
5. One contributor from each team should fork the repository and other contributors should mutually agree on working collaboratively under that forked repo. 

> 6. Under the forked repo, contributors should create an issue (i.e. the sub-topic name) and that issue link should be referenced in that corresponding sub-topic section in the main chapter outline. So, that the main admins can also get updated on status of each chapters and topics. 

Author: Yeah make sense. Okay then I should change this part right?

> 6. Under the forked repo, contributors should create an issue (i.e. the sub-topic name) and that issue link should be referenced in that corresponding sub-topic section in the main chapter outline. So, that the main admins can also get updated on status of each chapters and topics.

> 6. Under the forked repo, contributors should create an issue (i.e. the sub-topic name) and that issue link should be referenced in that corresponding sub-topic section in the main chapter outline. So, that the main admins can also get updated on status of each chapters and topics. 

Reviewer: Yes, exactly

7. After each sub-topics is finished by a team member, other members will review and approve it. And then they can push it the actual main repo. 
8. When contributor works on a subtopic, then they should follow the below instructions:
   1. Create `.mdx` files or Jupyter Notebooks for the sub-topics you want to contribute to
   2. Make sure to update the requirements.txt file in the root of the repository
   3.  When you feel like you are ready, create a pull request to this repository. 

>    4.  Once a PR is ready, let one the contributor (who will be in-charge of the forked repo) know about this. The contributor should after that update the status of outline by referencing that sub-topic with that PR. So that the admin can also be notified and updated on progress. So finally each sub-topic inside the chapter outline should look like this `sub-topic name | issue-link | PR-link`. Example: `Fine-tuning Vision Transformer for object detection | issue #x PR #y`

Reviewer: ```suggestion
   4.  Once a PR is ready, let the owner of the forked repo know about this. The owner should then update the status of the outline by referencing that sub-topic with that PR. So that the outline contains the current status of progress. Finally each sub-topic inside the chapter outline should look like this `sub-topic name | issue-link | PR-link`. Example: `Fine-tuning Vision Transformer for object detection | issue #x PR #y`
```

>    5.  Your teammates will review your PR under that forked repo and then if things get approve, you can create the PR to merge it our main branch 🤗

Reviewer: ```suggestion
   5.  Your teammates will review your PR under that forked repo and then if things get approved, you can create the PR to merge it in our main branch 🤗
```

 
2. Now any one or group of contributors can discuss and should come up with a granular structure of table of contents. For example, consider the Chapter 3: Vision Transformers.
   1. So, we create an issue for Vision Transformers.
   2. Under this, we can create three issues on main topics each (Transformers Architecture, Common ViTs, Transfer Learning and Fine-tuning ViTs)
   3. In the last level of granularity, under each topics there can be some bunch of sub-topics, [here is an example](https://github.com/johko/computer-vision-course/issues/53#issuecomment-1837081001).
   4. Now not going through this recursive loop, each sub-topics should be an atomic issue, assigned to some one. That atomic issue, should also reference the topic issue and the chapter issue. An example:
      1. Suppose someone wants to contribute on how fine-tuning is done on object detection. So the contributor creates an issue on "Fine-tuning ViTs for object detection"
      2. Inside that issue, the contributor must reference the topic issue i.e. `Transfer Learning and Fine-tuning ViTs` and the chapter issue: `Vision Transformers`, so that we can tag those accordingly.
   

> 3. **🚧 For this part, I am not sure.** Each contributor should work on a sub-topic level. Each topic should follow a branch naming convention. For example: `vit/tl-ft/object-detection`. The contributor forks the repo and makes a branch and simply works on that sub-topic. Once done, in the PR, it should not be merged to `main`. Rather that PR should be a part of a sub-branch. Example: `main:vit/tl-ft` < - `merveenoyan:vit/tf-ft/object-detection`. (Where `tl-ft` is transfer-learning and fine-tuning). Once, every sub-topics are merged, then the maintainers will merge `main:vit/tl-ft` to main.

Reviewer 2: I think this part is a bit too much work, I think it's ok if people just contribute and solve merge conflicts in a way that it doesn't break the flow of the material. WDYT?

Author: > That way, we might be able to have a team with the core people who have a general idea of how that topic is structured and where things are going, and we have the flexibility for the people to come in and improve a subtopic.

Yes, I agree on that 100%

   
    **Note:** The contributors, simply needs to work on their PR under their fork. They are not required to put additional headaches on understanding and maintaining branch structures necessarily 😊. That can be handled by course maintainers. (if this structure get's approved, we can make a small video on how to easily get started to contributing through a video 🤗)

> 4. Create `.mdx` files or Jupyter Notebooks for the sub-topics you want to contribute to

Reviewer 2: I think it will be a bit of work to always patch chapters together because each mdx file is rendered on it's own

Author: Oh, did't knew about that, any suggestions?
